prov/verbs/src/verbs_rma.c

```diff
@@ -131,41 +131,43 @@ fi_ibv_msg_ep_rma_readv(struct fid_ep *ep_fid, const struct iovec *iov, void **d
 		     uint64_t addr, uint64_t key, void *context)
 {
 	struct fi_ibv_msg_ep *ep;
-	struct ibv_send_wr wr;
-	size_t len = 0;
+	struct ibv_send_wr wr = { 0 };
+
+	ep = container_of(ep_fid, struct fi_ibv_msg_ep, ep_fid);
 
-	memset(&wr, 0, sizeof(wr));
 	wr.opcode = IBV_WR_RDMA_READ;
 	wr.wr.rdma.remote_addr = addr;
 	wr.wr.rdma.rkey = (uint32_t) key;
-
-	ep = container_of(ep_fid, struct fi_ibv_msg_ep, ep_fid);
 	wr.send_flags = VERBS_COMP_READ(ep);
+	wr.num_sge = count;
+
+	wr.sg_list = alloca(sizeof(*(wr.sg_list)) * count);
 
-	fi_ibv_set_sge_iov(wr.sg_list, iov, count, desc, len);
+	fi_ibv_iov_to_sgl(iov, wr.sg_list, count, desc);
```

Comment:
Using a macro instead of a function avoids repeating the sg_list allocation in every send call. That was the original intent.